### PR TITLE
mw/kubernetes: remove zone from parseRequest

### PR DESCRIPTION
**Pull request by miekg**
State has the zone info as well, so don't need to have it in
parseRequest anymore.

### CODE REVIEW

**Comment by miekg** [comment]
I'll double check why test coverage is going down. Could be just the deleted lines.

File: middleware/kubernetes/parse.go [hunk 1]
```diff
@@ -101,6 +98,5 @@ func (r recordRequest) String() string {
 	s += "." + r.service
 	s += "." + r.namespace
 	s += "." + r.podOrSvc
-	s += "." + r.zone
```
Comment (chrisohaver) [inline middleware/kubernetes/parse.go:104]: This function will escape to stack.  Consider using strings.Join() instead of "+"

**Review by chrisohaver — CHANGES_REQUESTED** [review]

File: middleware/kubernetes/parse.go (hunk 1 above)
Comment (miekg) [reply middleware/kubernetes/parse.go:104]: sorry, what? You mean each s is a new string and therefor adds GC?
Sure I can change this.

Comment (miekg) [reply middleware/kubernetes/parse.go:104]: strings.Join requires a string slice to be constructed. If you worried about mem...

Comment (chrisohaver) [reply middleware/kubernetes/parse.go:104]: Oh ok, never mind then, i must be mistaken.

**Review by chrisohaver — APPROVED** [review]

**Comment by miekg** [comment]
Travis 1.7 build failed with
~~~
$ docker run -d --net=host --name=etcd quay.io/coreos/etcd:v$ETCD_VERSION
Unable to find image 'quay.io/coreos/etcd:v2.3.1' locally
docker: Error response from daemon: Get https://quay.io/v2/coreos/etcd/manifests/v2.3.1: Get https://quay.io/v2/auth?scope=repository%3Acoreos%2Fetcd%3Apull&service=quay.io: net/http: TLS handshake timeout.
See 'docker run --help'.
~~~

**Pull request merged**